src/sqlfluff/utils/reflow/reindent.py

```diff
@@ -27,3 +36,792 @@ def deduce_line_indent(raw_segment: RawSegment, root_segment: BaseSegment) -> st
+    forced_indents: List[int],
+) -> List[LintFix]:
+    """Evalute a single set of indent points on one line.
+
+    NOTE: This mutates the given `elements` and `forced_indents` input to avoid
+    lots of copying.
+
+    Order of operations:
+    1. Evaluate the starting indent for this line.
+    2. For point which aren't line breaks in the line, we evaluate them
+       to see whether they *should* be. We seperately to missing indents
+       on the way *up* (i.e. where there's more indents than dedents)
+       and then on the way *down* (i.e. where there's more dedents than
+       indents).
+
+    This method returns fixes, including appropriate descriptions, to
+    allow generation of LintResult objects directly from them.
+    """
+    reflow_logger.debug(
+        "  Evaluate Line #%s [source line #%s]. FI %s",
+        elements[indent_line.indent_points[0].idx + 1]
+        .segments[0]
+        .pos_marker.working_line_no,
+        elements[indent_line.indent_points[0].idx + 1]
+        .segments[0]
+        .pos_marker.source_position()[0],
+        forced_indents,
+    )
+    reflow_logger.debug(
+        "   Line Segments: %s",
+        [
+            elem.segments
+            for elem in elements[
+                indent_line.indent_points[0].idx : indent_line.indent_points[-1].idx
+            ]
+        ],
+    )
+    reflow_logger.info("  Evaluate Line: %s. FI %s", indent_line, forced_indents)
+    fixes = []
+
+    # Catch edge case for first line where we'll start with a block if
+    # no initial indent.
+    # NOTE: We use the first point for the starting balance rather than
+    # the line starting balance because we're using this to detect missing
+    # lines and if the line has been corrected then we don't want to do
+    # that.
+    indent_points = indent_line.indent_points
+    # Set up the default anchor
+    anchor = {"before": elements[indent_points[0].idx + 1].segments[0]}
+
+    # Find initial indent, and deduce appropriate string indent.
+    # We also account for potential templated indents.
+    indent_seg = None
+    if indent_points[-1].last_line_break_idx:
+        indent_seg = cast(
+            ReflowPoint, elements[indent_points[-1].last_line_break_idx]
+        )._get_indent_segment()
+    elif isinstance(elements[0], ReflowPoint):
+        # No last_line_break_idx, but this is a point. It's the first line.
+        # Get the last whitespace element.
+        # TODO: We don't currently handle the leading swallowed whitespace case.
+        # That could be added here, but it's an edge case which can be done
+        # at a later date easily. For now it won't get picked up.
+        for indent_seg in elements[0].segments[::-1]:
+            if indent_seg.is_type("whitespace"):
+                break
+        # Handle edge case of no whitespace, but with newline.
+        if not indent_seg.is_type("whitespace"):
+            indent_seg = None
+        else:
+            NotImplementedError(
```

Comment:
I think you want to _raise_ the error, otherwise execution will continue below.
```suggestion
            raise NotImplementedError(
```

Reply:
GREAT CATCH. I found another instance of this elsewhere in the code while debugging and it took me _ages_ to debug.